plugins/media/wowza/services/LiveConversionProfileService.php

```diff
@@ -174,16 +192,107 @@ public function serveAction($streamName, $hostname = null)
 		
 		return new kRendererString($dom->saveXML(), 'text/xml');
 	}
-
-	protected function appendLiveParams(LiveStreamEntry $entry, WowzaMediaServerNode $mediaServer = null, SimpleXMLElement $encodes, liveParams $liveParams)
+	
+	private function isValidJson($strJson)
+	{
+		json_decode($strJson);
+		return (json_last_error() === JSON_ERROR_NONE);
+	}
+	
+	private function calculateFlavorHeight($flavorResolution, $ingestParameters)
+	{
+		return ($ingestParameters['width'] != 0) ? (($flavorResolution['width'] * $ingestParameters['height']) / $ingestParameters['width']) : 0;
```

Comment:
the strings width | hight | match-source are returning few times in the code why not use constants instead of strings?